### PR TITLE
Fix server for deployment to Zeit

### PR DESCRIPTION
**Pull request by flybayer**
Zeit automatically adds the serverless target to next.config.js, so it's always present during Zeit build time. Blitz build was yelling because that file shouldn't be there.

And we do actually need that config that Zeit sets, so this will preserve any config Zeit sets.

This is done except the test is failing and I can't figure out why. @ryardley could you take a look at it?

### CODE REVIEW

**Comment by ryardley** [comment]
Ok. Where does the NOW_BUILD env var come from? Is that a zeit thing?

**Comment by flybayer** [comment]
Yes, Zeit always provides it during build time

File: packages/server/src/synchronizer/rules/config.ts [hunk 1]
```diff
@@ -25,7 +25,20 @@ export async function copyConfig(entries: string[], srcPath: string, destPath: s
 
   await writeFile(resolve(destPath, 'blitz.config.js'), Buffer.from(fileContents))
 
-  const nextConfigShellTpl = `
+  // Zeit now adds configuration needed for Now, like serverless target,
+  // so we need to keep and use that
+  if (process.env.NOW_BUILDER) {
+    rename(resolve(srcPath, 'next.config.js'), resolve(destPath, 'next-zeit.config.js'))
```
Comment (ryardley) [inline packages/server/src/synchronizer/rules/config.ts:31]: Problem is you are missing an await here I think

Comment (ryardley) [reply packages/server/src/synchronizer/rules/config.ts:31]: Also we can't rename here. We can't alter the source folder as this is run after the entries have been taken.

Comment (ryardley) [reply packages/server/src/synchronizer/rules/config.ts:31]: Will push a fix

**Comment by flybayer** [comment]
@ryardley great, thank you!!

**Pull request merged**